Modules/_testbuffer.c

```diff
@@ -2813,48 +2813,27 @@ static struct PyMethodDef _testbuffer_functions[] = {
     {NULL, NULL}
 };
 
-static struct PyModuleDef _testbuffermodule = {
-    PyModuleDef_HEAD_INIT,
-    "_testbuffer",
-    NULL,
-    -1,
-    _testbuffer_functions,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-
-PyMODINIT_FUNC
-PyInit__testbuffer(void)
+static int
+_testbuffer_exec(PyObject *m)
```

Comment:
Converting _testbuffer to multiphase init allows to create two instances of the module, but the C implementation uses global variables which is incompatible with that:

```
static PyObject *Struct = NULL;
static PyObject *calcsize = NULL;
```

You must first avoid these global variables. I suggest you to attempt to pass the module to functions using Struct and calcsize, and then get Struct and calcsize from the module. Either use PyObject_GetAttrString() which can slow down the module (which is ok, the module is only used for testing purpose), or use a module state (see my other comment about that).

Reply:
@vstinner I am more comfortable with module state, anyways eventually it will need it if we migrate to heap types.